src/Tasks/AssemblyDependency/Resolver.cs

```diff
@@ -181,7 +196,27 @@ ResolutionSearchLocation searchLocation
 
             bool isSimpleAssemblyName = assemblyName?.IsSimpleName == true;
 
-            if (fileExists(pathToCandidateAssembly))
+            bool fileFound;
+            if (useDirectoryCache && Utilities.ChangeWaves.AreFeaturesEnabled(Utilities.ChangeWaves.Wave16_10))
+            {
+                // this verifies file existence using getDirectoryFile delegate which internally used cached list of all files in a particular directory
+                // if some cases it render better performance than one by one FileExists
+                try
+                {
+                    fileFound = getDirectoryFile(directory, fileName) != null;
```

Comment:
Can you remind me why this call takes directory + filename as opposed to full path? Is it an optimization where we don't want to burn cycles splitting the full path when we already have the two components? Or would something break functionally if the implementation was abstracted away and the signature was simply `bool (string)`?

Reply:
It is optimization. `bool (string)` would require Path.GetFileName and Path.GetDirectoryName with two related string allocations. Since this is known and initialized upper in call stack, I have decided to pass it down for perf reasons.